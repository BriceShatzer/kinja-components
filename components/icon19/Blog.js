// @flow

import * as React from 'react';

import Icon19 from './icon19';
import type { Icon19Type } from './icon19';
import Blog from './svg/Blog/Blog.svg';

const BlogIcon = (props: Icon19Type) =>
	<Icon19 {...props}>
		<Blog/>
	</Icon19>;

export default BlogIcon;
