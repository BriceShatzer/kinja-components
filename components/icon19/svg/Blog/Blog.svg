<svg xmlns="http://www.w3.org/2000/svg" width="18" height="18" aria-label="Blog icon" viewBox="0 0 18 18">
	<path fill-rule="evenodd" d="M17 14.5a2.5 2.5 0 0 1-2.5 2.5h-11A2.5 2.5 0 0 1 1 14.5v-11A2.5 2.5 0 0 1 3.5 1h11A2.5 2.5 0 0 1 17 3.5v11zM16 5H2v9.5c0 .83.67 1.5 1.5 1.5h11c.83 0 1.5-.67 1.5-1.5V5zm-1.5-3h-11C2.66 2 2 2.67 2 3.5V4h14v-.5c0-.83-.67-1.5-1.5-1.5zM5 12v1h1v-1H5zm-1 0a1 1 0 0 1 1-1h1a1 1 0 0 1 1 1v1a1 1 0 0 1-1 1H5a1 1 0 0 1-1-1v-1zm4.5 0a.5.5 0 1 1 0-1h5a.5.5 0 1 1 0 1h-5zm0 2a.5.5 0 1 1 0-1h3a.5.5 0 1 1 0 1h-3zM5 8v1h1V8H5zM4 8a1 1 0 0 1 1-1h1a1 1 0 0 1 1 1v1a1 1 0 0 1-1 1H5a1 1 0 0 1-1-1V8zm4.5 0a.5.5 0 0 1 0-1h5a.5.5 0 1 1 0 1h-5zm0 2a.5.5 0 0 1 0-1h3a.5.5 0 1 1 0 1h-3z"/>
</svg>
