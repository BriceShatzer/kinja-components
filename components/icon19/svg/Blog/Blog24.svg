<svg xmlns="http://www.w3.org/2000/svg" width="24" height="24" aria-label="Blog24 icon" viewBox="0 0 24 24">
	<path fill-rule="evenodd" d="M20.46 1C21.86 1 23 2.13 23 3.54v16.92c0 1.4-1.13 2.54-2.54 2.54H3.54A2.53 2.53 0 0 1 1 20.46V3.54C1 2.14 2.13 1 3.54 1zm1.52 5H2.02v14.46c0 .8.6 1.45 1.38 1.52h17.06c.85 0 1.52-.68 1.52-1.52V6zM6.93 16c.55 0 1 .41 1.06.95l.01.12v1.86c0 .55-.41 1-.95 1.06l-.12.01H5.07c-.55 0-1-.41-1.06-.95L4 18.93v-1.86c0-.55.41-1 .95-1.06l.12-.01h1.86zm-.47 1h-.92c-.3 0-.54.24-.54.54v.92c0 .3.24.54.54.54h.92c.3 0 .54-.24.54-.54v-.92c0-.3-.24-.54-.54-.54zm8.04 1a.5.5 0 0 1 .09 1H9.5a.5.5 0 0 1-.09-1h5.09zm5-2a.5.5 0 0 1 .09 1H9.5a.5.5 0 0 1-.09-1H19.5zM6.93 9c.55 0 1 .41 1.06.95l.01.12v1.86c0 .55-.41 1-.95 1.06l-.12.01H5.07c-.55 0-1-.41-1.06-.95L4 11.93v-1.86c0-.55.41-1 .95-1.06L5.07 9h1.86zm-.47 1h-.92c-.3 0-.54.24-.54.54v.92c0 .3.24.54.54.54h.92c.3 0 .54-.24.54-.54v-.92c0-.3-.24-.54-.54-.54zm8.04 1a.5.5 0 0 1 .09 1H9.5a.5.5 0 0 1-.09-1h5.09zm5-2a.5.5 0 0 1 .09 1H9.5a.5.5 0 0 1-.09-1H19.5zm.96-6.98H3.54c-.85 0-1.52.68-1.52 1.52V5h19.96V3.54c0-.8-.6-1.45-1.38-1.52h-.14z"/>
</svg>
