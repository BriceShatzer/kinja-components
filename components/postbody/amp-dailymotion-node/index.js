// @flow

import AmpDailyMotionNode from './amp-dailymotion-node';

export default AmpDailyMotionNode;
