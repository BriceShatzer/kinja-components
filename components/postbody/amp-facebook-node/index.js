// @flow

import AmpFacebookNode from './amp-facebook-node';
import AmpFacebookPostNode from './amp-facebookPost-node';

export {
	AmpFacebookNode,
	AmpFacebookPostNode
};