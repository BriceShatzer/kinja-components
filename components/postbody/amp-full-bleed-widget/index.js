// @flow

import AmpFullBleedWidget from './amp-full-bleed-widget';

export default AmpFullBleedWidget;
