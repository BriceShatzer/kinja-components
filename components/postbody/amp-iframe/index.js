// @flow

import AmpIframe from './amp-iframe';

export default AmpIframe;
