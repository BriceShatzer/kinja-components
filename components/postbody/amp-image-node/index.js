// @flow

import AmpImageNode from './amp-image-node';

export default AmpImageNode;
