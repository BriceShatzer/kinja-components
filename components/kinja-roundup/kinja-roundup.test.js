/* @flow */

import * as React from 'react';
import { shallow, mount } from 'enzyme';

import KinjaRoundup from './kinja-roundup';

import stubbedPost from 'kinja-components/__stubs__/stubbedPost.json';
import stubbedBlog from 'kinja-components/__stubs__/stubbedBlog.2.json';
import Post from 'kinja-magma/models/Post';
import Blog from 'kinja-magma/models/Blog';

import { TextNode, LinkNode } from 'postbody/InlineNode';
import { RoundupItem } from 'postbody/blockNodes/Roundup';

const externalAPI = {
	getPost: () => new Promise(resolve => resolve(Post.fromJSON(stubbedPost))),
	getBlog: () => new Promise(resolve => resolve(Blog.fromJSON(stubbedBlog))),
	updatePostModel: () => {}
};

const props = {
	data: [
		new RoundupItem(
			new LinkNode(
				[new TextNode('Ditch The Dating Apps: 5 Real-Life Places To Meet My Wife')],
				'https://www.clickhole.com/ditch-the-dating-apps-5-real-life-places-to-meet-my-wi-1832472134'
			),
			new TextNode('ClickHole')
		),
		new RoundupItem(
			new LinkNode(
				[new TextNode('Tips For Playing Anthem')],
				'https://kotaku.com/tips-for-playing-anthem-1832821302'
			),
			new TextNode('Kotaku')
		)
	]
};

describe('<KinjaRoundup />', () => {
	it('should render by default', () => {
		const wrapper = shallow(<KinjaRoundup
			externalAPI={externalAPI}
			{...props} />);
		expect(wrapper).toMatchSnapshot();
	});

	it('should call getPost when inserted', () => {
		const getPostHandler = jest.fn();
		externalAPI.getPost = getPostHandler;

		const roundup = mount(<KinjaRoundup externalAPI={externalAPI} data={[]}/>);
		expect(roundup.find('input').length).toEqual(5);

		// $FlowFixMe
		roundup.find('input').first().getDOMNode().value = 'https://kotaku.com/tips-for-playing-anthem-1832821302';
		roundup.find('input').first().simulate('change');

		expect(externalAPI.getPost).toHaveBeenCalledWith({id: 1832821302});
	});

	it('saveItems should call updatePostModel with the correct items', () => {
		const handler = jest.fn();
		externalAPI.updatePostModel = handler;

		const roundup = mount(<KinjaRoundup externalAPI={externalAPI} {...props} />);
		// $FlowFixMe
		roundup.instance().saveItems();

		expect(externalAPI.updatePostModel).toHaveBeenCalledWith(props.data);
	});
});
