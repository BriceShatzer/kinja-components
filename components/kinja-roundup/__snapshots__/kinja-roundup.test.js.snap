// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<KinjaRoundup /> should render by default 1`] = `
<Fragment>
  <kinja-roundup__RoundupWrapper
    borderbottom={true}
  >
    <kinja-roundup__RoundupItemsWrapper>
      <kinja-roundup__RoundupHelpText>
        Paste a Kinja permalink URL into each field to create a Roundup.
      </kinja-roundup__RoundupHelpText>
      <kinja-roundup__RoundupItems>
        <div
          key="-332065034"
        >
          <kinja-roundup__RoundupItemRow>
            <Textfield18
              blogTheme="default"
              data-index={0}
              error=""
              key="-332065034"
              label="Story 1"
              name="Roundup Item"
              onChange={[Function]}
              value="Ditch The Dating Apps: 5 Real-Life Places To Meet My Wife"
            />
            <kinja-roundup__DeleteItem
              data-index={0}
              onClick={[Function]}
            >
              <TrashcanIcon />
            </kinja-roundup__DeleteItem>
          </kinja-roundup__RoundupItemRow>
        </div>
        <div
          key="647052003"
        >
          <kinja-roundup__RoundupItemRow>
            <Textfield18
              blogTheme="default"
              data-index={1}
              error=""
              key="647052003"
              label="Story 2"
              name="Roundup Item"
              onChange={[Function]}
              value="Tips For Playing Anthem"
            />
            <kinja-roundup__DeleteItem
              data-index={1}
              onClick={[Function]}
            >
              <TrashcanIcon />
            </kinja-roundup__DeleteItem>
          </kinja-roundup__RoundupItemRow>
        </div>
        <div
          key="2"
        >
          <kinja-roundup__RoundupItemRow>
            <Textfield18
              blogTheme="default"
              data-index={2}
              error=""
              key="2"
              label="Story 3"
              name="Roundup Item"
              onChange={[Function]}
              value=""
            />
            <kinja-roundup__DeleteItem
              data-index={2}
              onClick={[Function]}
            >
              <TrashcanIcon />
            </kinja-roundup__DeleteItem>
          </kinja-roundup__RoundupItemRow>
        </div>
      </kinja-roundup__RoundupItems>
    </kinja-roundup__RoundupItemsWrapper>
    <hr />
    <kinja-roundup__RoundupPreviewWrapper>
      <RoundupComponent
        hideBlogInfo={false}
        intro={Array []}
        items={
          Array [
            RoundupItem {
              "blog": TextNode {
                "styles": Array [],
                "type": "Text",
                "value": "ClickHole",
              },
              "link": LinkNode {
                "reference": "https://www.clickhole.com/ditch-the-dating-apps-5-real-life-places-to-meet-my-wi-1832472134",
                "target": undefined,
                "type": "Link",
                "value": Array [
                  TextNode {
                    "styles": Array [],
                    "type": "Text",
                    "value": "Ditch The Dating Apps: 5 Real-Life Places To Meet My Wife",
                  },
                ],
              },
              "type": "RoundupItem",
            },
            RoundupItem {
              "blog": TextNode {
                "styles": Array [],
                "type": "Text",
                "value": "Kotaku",
              },
              "link": LinkNode {
                "reference": "https://kotaku.com/tips-for-playing-anthem-1832821302",
                "target": undefined,
                "type": "Link",
                "value": Array [
                  TextNode {
                    "styles": Array [],
                    "type": "Text",
                    "value": "Tips For Playing Anthem",
                  },
                ],
              },
              "type": "RoundupItem",
            },
          ]
        }
      />
    </kinja-roundup__RoundupPreviewWrapper>
  </kinja-roundup__RoundupWrapper>
</Fragment>
`;
